exporters/stdout/stdoutlog/exporter.go

```diff
@@ -33,26 +54,94 @@ func New(options ...Option) (*Exporter, error) {
 		timestamps: cfg.Timestamps,
 	}
 	e.encoder.Store(enc)
+	e.initSelfObservability()
 
 	return &e, nil
 }
 
+// initSelfObservability initializes self-observability for the exporter if enabled.
+func (e *Exporter) initSelfObservability() {
+	if !x.SelfObservability.Enabled() {
+		return
+	}
+
+	e.selfObservability = &selfObservability{
+		enabled: true,
+		attrs: []attribute.KeyValue{
+			semconv.OTelComponentName(fmt.Sprintf("%s/%d", otelComponentType, nextExporterID())),
+			semconv.OTelComponentTypeKey.String(otelComponentType),
+		},
+	}
+
+	mp := otel.GetMeterProvider()
+	m := mp.Meter("go.opentelemetry.io/otel/exporters/stdout/stdoutlog",
+		metric.WithInstrumentationVersion(sdk.Version()),
+		metric.WithSchemaURL(semconv.SchemaURL),
+	)
+
+	var err error
+	if e.selfObservability.inflightMetric, err = otelconv.NewSDKExporterLogInflight(m); err != nil {
+		otel.Handle(err)
+	}
+	if e.selfObservability.exportedMetric, err = otelconv.NewSDKExporterLogExported(m); err != nil {
+		otel.Handle(err)
+	}
+	if e.selfObservability.operationDurationMetric, err = otelconv.NewSDKExporterOperationDuration(m); err != nil {
+		otel.Handle(err)
+	}
+}
+
 // Export exports log records to writer.
 func (e *Exporter) Export(ctx context.Context, records []log.Record) error {
 	enc := e.encoder.Load()
 	if enc == nil {
 		return nil
 	}
 
+	var err error
+	if e.selfObservability != nil && e.selfObservability.enabled {
+		err = e.exportWithSelfObservability(ctx, records)
+	} else {
+		err = e.exportWithoutSelfObservability(ctx, records)
+	}
+	return err
+}
+
+// exportWithSelfObservability exports logs with self-observability metrics.
+func (e *Exporter) exportWithSelfObservability(ctx context.Context, records []log.Record) error {
+	if len(records) == 0 {
+		return nil
+	}
+
+	count := int64(len(records))
+	start := time.Now()
+
+	e.selfObservability.inflightMetric.Add(context.Background(), count, e.selfObservability.attrs...)
+	defer func() {
+		addAttrs := make([]attribute.KeyValue, len(e.selfObservability.attrs), len(e.selfObservability.attrs)+1)
```

Comment:
This is allocated every call. A pool should be used to amortize the slice allocation.